src/dbnode/storage/namespace/context.go

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package namespace
+
+import (
+	"github.com/m3db/m3/src/x/ident"
+)
+
+type Context struct {
+	Id     ident.ID
+	Schema SchemaDescr
+}
+
+func NewContextFrom(nsMetadata Metadata) Context {
+	ctx := Context{Id: nsMetadata.ID()}
+	if schema, ok := nsMetadata.Options().SchemaHistory().GetLatest(); ok {
```

Comment:
Does this mean that during bootstrap all encoders/iterators will used a "stale" schema if the schema changes AFTER the bootstrap starts? I think thats probably fine but I'm just trying to make sure I understand properly.

Reply:
yes, your understanding is correct.